lib.nix

```diff
@@ -146,6 +149,12 @@ let
       in
       eval.config.flake;
 
+    # Shorthand for `types.submoduleWith`.
+    submoduleWithModules =
+      { ... }@attrs:
+      modules:
+      types.submoduleWith (attrs // { modules = toList modules; });
```

Comment:
This doesn't really do anything new, but it does make flake-parts a dialect of the module system. I would like to avoid that.
All functions in this library should either be polyfills of Nixpkgs lib, or domain-specific helpers (e.g. supporting perSystem).

Reply:
Done. I should submit this up to Nixpkgs for review soon, I think it fits in well as an alternative to `submodule` / `deferredModule` that offers finer control. It's still defined as an internal local in `nixpkgs.nix` because otherwise the indentation gets really intense.